### PR TITLE
 routematch: Remove deprecated runtime field

### LINKED ISSUE #4623: [v1.8.0 (Oct 4, 2018) deprecation] Remove features marked deprecated in #4560
#4560 (route match: Add runtime_fraction field for more granular routing) introduced a deprecation notice for v1.8.0 (Oct 4, 2018). This issue tracks source code cleanup.

### PR DESCRIPTION
**Pull request by tonya11en**
*Description*:
This patch removes the deprecated `runtime` field from the RouteMatch proto.

*Risk Level*:
Low.

*Testing*:
Unit tests and build.

*Docs Changes*:
Done.

Fixes #4623

### CODE REVIEW

**Comment by tonya11en** [comment]
@ccaraman tests are passing. I'll merge from master to pick up the ipv6 test after the first round of comments.

File: source/common/config/rds_json.cc [hunk 1]
```diff
@@ -224,10 +224,6 @@ void RdsJson::translateRoute(const Json::Object& json_route, envoy::api::v2::rou
 
   JSON_UTIL_SET_BOOL(json_route, *match, case_sensitive);
 
-  if (json_route.hasObject("runtime")) {
```
Comment (mattklein123) [inline source/common/config/rds_json.cc]: I don't think this is right. I think you need to translate to runtime fraction. Please make sure there is also a test that covers this.

**Comment by junr03** [comment]
@tonya11en this will also require us updating our internal configuration to make sure we dont depend on the deprecated field. I have been thinking about creating a process around this in our team. Lets chat when you get in today.

**Comment by tonya11en** [comment]
@junr03 I already modified the internal configs prior to sending out this PR. Happy to chat more today.

**Pull request merged**